stdlib/argparse.pyi

```diff
@@ -33,25 +33,14 @@ _ActionT = TypeVar("_ActionT", bound=Action)
 _ArgumentParserT = TypeVar("_ArgumentParserT", bound=ArgumentParser)
 _N = TypeVar("_N")
 _ActionType: TypeAlias = Callable[[str], Any] | FileType | str
-# more precisely, Literal["store", "store_const", "store_true",
-# "store_false", "append", "append_const", "count", "help", "version",
-# "extend"], but using this would make it hard to annotate callers
-# that don't use a literal argument
-_ActionStr: TypeAlias = str
-# more precisely, Literal["?", "*", "+", "...", "A...",
-# "==SUPPRESS=="], but using this would make it hard to annotate
-# callers that don't use a literal argument
-_NArgsStr: TypeAlias = str
 
 ONE_OR_MORE: Final = "+"
 OPTIONAL: Final = "?"
 PARSER: Final = "A..."
 REMAINDER: Final = "..."
-_SUPPRESS_T = NewType("_SUPPRESS_T", str)
-SUPPRESS: _SUPPRESS_T | str  # not using Literal because argparse sometimes compares SUPPRESS with is
-# the | str is there so that foo = argparse.SUPPRESS; foo = "test" checks out in mypy
+SUPPRESS: Final = "==SUPPRESS=="
```

Comment:
This comment is obsolete, both mypy and pyright are happy with this idiom
```python
import argparse
reveal_type(argparse.SUPPRESS)  # reveal "Literal['==SUPPRESS==']"
help = argparse.SUPPRESS
help = "help"
```

Also pyright recently learned how to correctly narrow with identity checking a literal (using `if help is argparse.SUPPRESS`), mypy already worked fine

Reply:
FYI This particular change might still have been required. Removing it broke the mypyc compiled test runs. See https://github.com/python/mypy/pull/18580.

I partially reverted this commit in https://github.com/python/mypy/pull/18683 -> https://github.com/python/mypy/pull/18683/commits/756e61ac53887a18e0131fb7f794b6b6c2fb8caa to fix the mypy typeshed sync.